microsetta_public_api/api/plotting.py

```diff
@@ -1,11 +1,76 @@
-def plot_alpha_filtered(alpha_metric, age_cat=None, bmi_cat=None,
-                        percentiles=None, sample_id=None):
-    raise NotImplementedError()
+import pandas as pd
+import altair as alt
+from microsetta_public_api.repo._alpha_repo import AlphaRepo
+from microsetta_public_api.repo._metadata_repo import MetadataRepo
+from microsetta_public_api.models._alpha import Alpha
+from microsetta_public_api.api.metadata import _format_query, \
+    _validate_query, _filter_matching_ids
+from microsetta_public_api.utils._utils import jsonify
+
+
+def plot_alpha_filtered(alpha_metric=None, percentiles=None,
+                        sample_id=None, **kwargs):
+    repo = MetadataRepo()
+    query = _format_query(kwargs)
+    is_invalid = _validate_query(kwargs, repo)
+    if is_invalid:
+        return is_invalid
+
+    return _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles,
+                                                query, repo, sample_id)
 
 
-def plot_alpha_filtered_json_query(body, alpha_metric, percentiles=None,
+def plot_alpha_filtered_json_query(body, alpha_metric=None, percentiles=None,
                                    sample_id=None):
-    raise NotImplementedError()
+    repo = MetadataRepo()
+
+    return _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles,
+                                                body, repo, sample_id)
+
+
+def _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles, query,
+                                         repo, sample_id):
+    matching_ids = repo.sample_id_matches(query)
+    matching_ids, error_code, error_response = _filter_matching_ids(
+        matching_ids, AlphaRepo, 'available_metrics', alpha_metric, 'metric',
+    )
+    if sample_id:
+        _, error_code, error_response = _filter_matching_ids(
+            [sample_id], AlphaRepo, 'available_metrics', alpha_metric,
+            'metric', error_code=error_code, error_response=error_response,
+        )
+    if error_response:
+        return error_response, error_code
+    if len(matching_ids) <= 1:
```

Comment:
for example, this and the error_response check above could be performed upstream of this method which would make this function a lot more focused